framework/src/modules/token/api.ts

```diff
@@ -52,55 +63,87 @@ export class TokenAPI extends BaseAPI {
 		apiContext: APIContext,
 		senderAddress: Buffer,
 		recipientAddress: Buffer,
-		_tokenID: TokenID,
+		tokenID: TokenID,
 		amount: bigint,
 	): Promise<void> {
+		const canonicalTokenID = await this._getCanonicalTokenID(apiContext, tokenID);
 		const userStore = apiContext.getStore(this.moduleID, STORE_PREFIX_USER);
-		const sender = await userStore.getWithSchema<UserStoreData>(senderAddress, userStoreSchema);
-		if (sender.availableBalance < amount + this._minBalance) {
+		const sender = await userStore.getWithSchema<UserStoreData>(
+			getUserStoreKey(senderAddress, canonicalTokenID),
+			userStoreSchema,
+		);
+		if (sender.availableBalance < amount) {
 			throw new Error(
 				`Sender ${senderAddress.toString(
 					'hex',
-				)} balance ${sender.availableBalance.toString()} is not sufficient for ${(
-					amount + this._minBalance
-				).toString()}`,
+				)} balance ${sender.availableBalance.toString()} is not sufficient for ${amount.toString()}`,
 			);
 		}
+
 		sender.availableBalance -= amount;
-		await userStore.setWithSchema(senderAddress, sender, userStoreSchema);
+
+		await userStore.setWithSchema(
```

Comment:
Here do we risk subtracting the amount from the sender, and then the API failing for min balance ?

Reply:
if the API fails, the store rollback, so it is ok to set the value